File: Lib/Models/Activity.cs
using System.Diagnostics.CodeAnalysis;

namespace Lib.Models;


[ExcludeFromCodeCoverage]
public class Activity : BaseModel
{
    public Guid ActivityId { get; set; }
    
    public Guid? CategoryId { get; set; }
    
    public Guid? JobId { get; set; }
    
    public DateTime StartTime { get; set; }
    
    public DateTime EndTime { get; set; }
    
    public string Name { get; set; }
    
    public string Description { get; set; }
}

public static class ActivityExtensions
{
    public static Activity UpdateWith(this Activity existing, Activity updated)
    {
        existing.Name = updated.Name;
        existing.Description = updated.Description;
        existing.StartTime = updated.StartTime;
        existing.EndTime = updated.EndTime;
        return existing;
    }
}

File: App/ClientApp/src/app/core/models/activity.ts
export interface Activity {
  activityId: string;
  categoryId: string | null;
  jobId: string | null;
  startTime: Date;
  endTime: Date;
  name: string;
  description: string;
  // BaseModel properties
  createdAt: Date;
  modifiedAt: Date;
  userId: string;
  concurrencyStamp: string;
}

export const defaultActivity: Activity = {
  activityId: "",
  categoryId: null,
  jobId: null,
  startTime: new Date(),
  endTime: new Date(),
  name: "",
  description: "",
  createdAt: new Date(),
  modifiedAt: new Date(),
  userId: "",
  concurrencyStamp: "",
};
